### PR TITLE
logexporter: Capture kubelet's /metrics at the end of tests

### PR DESCRIPTION
**Pull request by smarterclayton**
/metrics is roughly ~120-150kb and often captures important metrics relating to overall test behavior that is useful for after-the-fact analysis and discovery (that could lead to new tests). Attempt to save the /metrics during log-dump phase.

Discussed briefly in slack https://kubernetes.slack.com/archives/C09QZ4DQB/p1677870251316299 with Ben

### CODE REVIEW

**Comment by smarterclayton** [comment]
Tested locally that the bash works and that I get metric output, not sure what sanity testing people need though beyond that.

**Comment by mborsz** [comment]
Do we have some rough estimate how much time will the additional dump take e.g. in 5000 node cluster?

File: logexporter/cluster/log-dump.sh
Comment (smarterclayton) [reply logexporter/cluster/log-dump.sh:837]: Capped at 5, agreed that's a reasonable start (and we can explore later broadening)

**Comment by SergeyKanzhelev** [comment]
So tests will need `USE_TEST_INFRA_LOG_DUMPING` to use this, correct? https://github.com/kubernetes/kubernetes/blob/88cce39155f3080cfe3163c9afb39b7572ca645c/cluster/log-dump/README.md?plain=1#L12

I think we have it on many jobs, but maybe we need to do more.

**Comment by SergeyKanzhelev** [comment]
BTW, browsing code I stumbled on `gather-metrics-at-teardown` [argument](https://github.com/kubernetes/kubernetes/blob/8319ac52747e0e0bd4ccb6cf9fb9ea168587c7b8/test/e2e/framework/test_context.go#LL335C56-L335C82). Would it do the same or similar thing? I cannot find anything that use this flag though

**Comment by smarterclayton** [comment]
Does test-infra support running candidate versions of jobs when you mutate them before merging?  Wasn't sure if steve had copied that for openshift or whether it was net new.  I added the flag to test to a node ci job in the commit to see.

**Comment by SergeyKanzhelev** [comment]
No, it doesn't. @bobbypage has shown me the way some time back to run them locally, but I never tried it. Maybe you can try to add this argument when you run these tests "locally".

Would you like to try it on some optional PR job that you can trigger from the PR review?

**Review by xmcqueen — COMMENTED** [review]
/lgtm

**Review by BenTheElder — COMMENTED** [review]
/lgtm
/approve
/hold

This looks reasonable to me with the cap on cluster size.

**Comment by BenTheElder** [comment]
Locally running works OK for things like unit tests but not so much for GCE e2e jobs because your local environment won't have the service account credentials or the boskos service to vend a project.

There are multiple options for locally testing jobs that don't need external resources:
https://docs.prow.k8s.io/docs/components/cli-tools/phaino/
https://github.com/kubernetes/test-infra/blob/master/config/pj-on-kind.sh

**Comment by smarterclayton** [comment]
Dropped the jenkins change, will test that as an alternative and potentially broaden that way at a later point (after i know what output it creates).  This is now just the previous change.

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm

Maybe you can leave a TODO in this method suggesting that it may be replaced at some point

**Comment by smarterclayton** [comment]
Heh:

```
I0313 15:35:51.024] ----------------------------------------------------------------------------------------------------
I0313 15:35:51.026] k/k version of the log-dump.sh script is deprecated!
I0313 15:35:51.026] Please migrate your test job to use test-infra's repo version of log-dump.sh!
I0313 15:35:51.026] Migration steps can be found in the readme file.
I0313 15:35:51.026] ----------------------------------------------------------------------------------------------------
```

https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/115357/pull-kubernetes-e2e-gce/1635291690187100160/build-log.txt

I'll take a look at that as well.

**Comment by SergeyKanzhelev** [comment]
> k/k version of the log-dump.sh script is deprecated!

see my comment above: https://github.com/kubernetes/test-infra/pull/28919#issuecomment-1459136980

Just add an env variable to the job

**Pull request merged**
